### ISSUE TITLE
Inconsistency between gradients and forces

### ISSUE THREAD

**Issue by raimis** [torchmd/torchmd-net#116]
When training with forces, the model computes the negative gradient of the energy with respect to the positions (a.k.a. forces):
https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/models/model.py#L196-L207

Some the loaders load forces:
https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/datasets/ani.py#L279-L281
https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/datasets/comp6.py#L100-L102

While the other ones load gradients (the opposite sign):
https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/datasets/qm9q.py#L147-L151
https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/datasets/spice.py#L99-L103

We need to agree what `y` represents.

**Comment by stefdoerr** [comment]
IMO `dy` should be the derivative of `y` i.e. the energy. Meaning the negative of the force. We should not store forces in `dy`

**Comment by raimis** [comment]
Ping: @PhilippThoelke @giadefa

**Comment by raimis** [comment]
Also checking the loss function code and comments, it seems `dy` is interpreted as gradients (not forces).

https://github.com/torchmd/torchmd-net/blob/db72e12461b1bce9bb3ddebd093fa11a803d37ca/torchmdnet/module.py#L71-L132

**Comment by PhilippThoelke** [comment]
The model outputs force predictions.
https://github.com/torchmd/torchmd-net/blob/main/torchmdnet/models/model.py#L207

I agree that the naming is inconsistent but as far as I remember, when I implemented QM9, MD17 and ANI-1, they all loaded forces instead of the derivative. The model was consistent with that.

I'm not sure about the more recently added dataset loaders.

**Comment by giadefa** [comment]
@peastman is SPICE returning forces or gradients?

            all_dy = (
                pt.tensor(mol["dft_total_gradient"], dtype=pt.float32)
                * self.HARTREE_TO_EV
                / self.BORH_TO_ANGSTROM
            )

**Comment by peastman** [comment]
I'm not sure.  Raimondas wrote that class.

**Comment by giadefa** [comment]
so you are not using that loader? Which one are you using?

**Comment by peastman** [comment]
HDF5.

**Comment by giadefa** [comment]
The datasets measure forces and I would expect that they store forces
